src/download.test.ts

```diff
@@ -29,50 +40,94 @@ Object.defineProperty(window, "location", {
 });
 
 describe("downloadFile", () => {
-  it("should call window.open with a download url with attachment set", () => {
-    validSkylinkVariations.forEach((input) => {
+  it("should download with a skylink url with attachment set", () => {
+    validSkylinkVariations.forEach(([fullSkylink, path]) => {
       mockLocationAssign.mockClear();
+      const url = client.downloadFile(fullSkylink);
+
+      let expectedUrl2 = `${expectedUrl}${path}${attachment}`;
+      if (path.startsWith("#")) {
+        // Hash should come after query.
+        expectedUrl2 = `${expectedUrl}${attachment}${path}`;
+      }
+      // Change ?attachment=true to &attachment=true if need be.
+      if ((expectedUrl2.match(/\?/g) || []).length > 1) {
+        expectedUrl2 = expectedUrl2.replace(attachment, "&attachment=true");
+      }
```

Comment:
I think that's too complicated logic for unit test to have, we should probably just stick with writing down the urls manually for each test case. Having regex and replace etc in a unit test sounds like that test could use a unit test to test that haha.

Reply:
I think it is ok here. It tests our URL-forming code without using the URL-forming library so it is like an extra check, using different logic.